File: Array/21_30/24Solution.py
def maxArea(h,w,horizontalCuts,verticalCuts):
        horizontalCuts=sorted(horizontalCuts)
        verticalCuts=sorted(verticalCuts)
        maxh=0
        maxv=0
        if (len(horizontalCuts)==1):
            maxh=max(horizontalCuts[0],h-horizontalCuts[0])
        else:
            for i in range(len(horizontalCuts)-1):
                maxh=max(maxh,horizontalCuts[i+1]-horizontalCuts[i])
        if(h-horizontalCuts[len(horizontalCuts)-1]>maxh or horizontalCuts[0]>maxh):
            maxh=max(h-horizontalCuts[len(horizontalCuts)-1],horizontalCuts[0])
        if (len(verticalCuts)==1):
            maxv=max(verticalCuts[0],w-verticalCuts[0])
        else:
            for i in range(len(verticalCuts)-1):
                maxv=max(maxv,verticalCuts[i+1]-verticalCuts[i])
        if(w-verticalCuts[len(verticalCuts)-1]>maxv or verticalCuts[0]>maxv):
            maxv=max(w-verticalCuts[len(verticalCuts)-1],verticalCuts[0])
        return(maxh*maxv%1000000007)
horizontalCuts=[ int(x.strip()) for x in input().split(",")]
verticalCuts=[ int(x.strip()) for x in input().split(",")]
h=int(input())
w=int(input())
print(maxArea(h,w,horizontalCuts,verticalCuts))

File: Array/21_30/24Solution.java
class Solution {
    public int maxArea(int h, int w, int[] horizontalCuts, int[] verticalCuts) {
        Arrays.sort(horizontalCuts);
        int lenH=horizontalCuts.length;
        
        int maxH=Math.max(horizontalCuts[0],h-horizontalCuts[lenH-1]);
        for(int i=1;i<horizontalCuts.length;i++){
            int diff=horizontalCuts[i]-horizontalCuts[i-1];
            if(maxH<diff){
                maxH=diff;
            }
        }
        
        Arrays.sort(verticalCuts);
        int lenV=verticalCuts.length;
        
        int maxV=Math.max(verticalCuts[0],w-verticalCuts[lenV-1]);
        for(int i=1;i<verticalCuts.length;i++){
            int diff=verticalCuts[i]-verticalCuts[i-1];
            if(maxV<diff){
                maxV=diff;
            }
        }
        return (int)((long)maxH * maxV % 1000000007);
    }
}
